docs/source/reference.rst

```diff
@@ -0,0 +1,176 @@
+Reference
+=========
+
+Trio mode
+---------
+
+Most users will want to enable "Trio mode". Without Trio mode:
+
+* Async tests have to be decorated with ``@pytest.mark.trio``
+* Async fixtures have to be decorated with
+  ``@pytest_trio.trio_fixture`` (instead of ``@pytest.fixture``).
```

Comment:
Oops, right, this text got confused – I originally wrote this thinking that async fixtures would always require either trio-mode or `trio_fixture`, then I switched to automatically coercing async fixtures used by `pytest.mark.trio`-tests, but I didn't update this text.

`@trio_fixture` is also useful to explicitly declare that a fixture is a trio fixture, so e.g. if you accidentally try to use it with a synchronous or asyncio test then you'll get a nice error message instead of some weird crash.

After sleeping on #51 and investigating some more, I think I actually want to switch the behavior again, to the union of the two things we've talked about: if trio-mode is enabled then all async fixtures are automatically converted to trio fixtures, *and* even when trio-mode is disabled then all async fixtures that are used by a trio test are automatically converted to trio fixtures. That keeps the current convenience for when we're not in non-trio-mode (and is backwards compatible), and then for projects that use trio-mode it also lets us catch mistakes like bad scopes or using async fixtures with a non-trio test.